exporter/datadogexporter/config/warning_deprecated.go

```diff
@@ -0,0 +1,97 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/config"
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/config"
+	"go.uber.org/multierr"
+)
+
+// List of settings that are deprecated.
+var renamedSettings = []renameError{
+	{
+		oldName:      "metrics::send_monotonic_counter",
+		newName:      "metrics::sums::cumulative_monotonic_mode",
+		oldRemovedIn: "v0.50.0",
+		issueNumber:  8489,
+		updateFn: func(c *Config) {
+			if c.Metrics.SendMonotonic {
+				c.Metrics.SumConfig.CumulativeMonotonicMode = CumulativeMonotonicSumModeToDelta
+			} else {
+				c.Metrics.SumConfig.CumulativeMonotonicMode = CumulativeMonotonicSumModeRawValue
+			}
+		},
+	},
+}
+
+var _ error = (*renameError)(nil)
+
+// renameError is an error related to a renamed setting.
+type renameError struct {
+	// oldName of the configuration option.
+	oldName string
+	// newName of the configuration option.
+	newName string
+	// oldRemovedIn is the version where the old config option will be removed.
+	oldRemovedIn string
+	// updateFn updates the configuration to map the old value into the new one.
+	// It must only be called when the old value is set and is not the default.
+	updateFn func(*Config)
+	// issueNumber on opentelemetry-collector-contrib for tracking
+	issueNumber uint
+}
+
+// Error implements the error interface.
+func (e renameError) Error() string {
+	return fmt.Sprintf(
+		"%q has been deprecated in favor of %q and will be removed in %s. See github.com/open-telemetry/opentelemetry-collector-contrib/issues/%d",
+		e.oldName,
+		e.newName,
+		e.oldRemovedIn,
+		e.issueNumber,
+	)
+}
+
+// Check if the deprecated option is being used.
+// Error out if both the old and new options are being used.
+func (e renameError) Check(configMap *config.Map) (bool, error) {
+	if configMap.IsSet(e.oldName) && configMap.IsSet(e.newName) {
+		return false, fmt.Errorf("%q and %q can't be both set at the same time: use %q only instead", e.oldName, e.newName, e.newName)
+	}
+	return configMap.IsSet(e.oldName), nil
+}
+
+// UpdateCfg to move the old configuration value into the new one.
+func (e renameError) UpdateCfg(cfg *Config) {
+	e.updateFn(cfg)
+}
+
+// handleRenamedSettings for a given configuration map.
+// Error out if any pair of old-new options are set at the same time.
+func handleRenamedSettings(configMap *config.Map, cfg *Config) (warnings []error, err error) {
+	for _, renaming := range renamedSettings {
+		ok, errCheck := renaming.Check(configMap)
```

Comment:
It's a bit weird for a boolean to be named `ok` if we log a warning when it's true 😄, but I'm not sure what would be a better fit here.

Reply:
I went with `isOldNameUsed`, it was weird indeed 😅